### PR TITLE
fix: rework internal filtering on versions

### PR DESCRIPTION
**Pull request by oklopfer**
tricky packages that all parse correctly now:
`neovim`
`github-cli-deb`
`firefox-arm64-deb`
`brave-browser-nightly-deb`

### CODE REVIEW

File: pacup [hunk 1]
```diff
@@ -205,28 +193,40 @@ sub repology_get_newestver ( $response, $filters, $oldver, $action ) {
...
+
     foreach my $entry (@$decoded) {
         next if grep /^\Q$entry->{'repo'}\E$/, @BANNED_REPOS;
         my $is_match
             = all { exists $entry->{$_} && $entry->{$_} eq $filters->{$_} }
             keys %$filters;
         next unless $is_match;
-        next if ( $entry->{'version'} eq 'HEAD' );
+        next if ($entry->{'version'} =~ /^(HEAD|main|master|scm|git|vcs|stable|9999)$/
+            || $entry->{'status'} eq 'rolling' && $filters->{'status'} ne 'rolling');
```
Comment (ghost) [inline pacup]:
```suggestion
        next if $entry->{'version'} =~ /^(HEAD|main|master|scm|git|vcs|stable|9999)$/
            || ( $entry->{'status'} eq 'rolling' && $filters->{'status'} ne 'rolling' );
```

Comment (oklopfer) [reply pacup]: too slow

Comment (ghost) [reply pacup]: No, my suggestion moves the parens

Comment (oklopfer) [reply pacup]: wait why

Comment (ghost) [reply pacup]: To make the precedence clear, because || and && have the same precedence. Even if it works here, the intention is not apparent

**Comment by ghost** [comment]
```
$ pacup packages/neovim/neovim.pacscript
[+] INFO: Parsing .SRCINFO for neovim.pacscript
    [>] Found pkgname: neovim
    [>] Found pkgver: 0.10.0
    [>] Found maintainer: Elsie19 <elsie19@pm.me>
    [>] Found repology info: project: neovim
[+] INFO: Querying Repology
** GET https://repology.org/api/v1/project/neovim ==> 200 OK
Use of uninitialized value in string ne at /home/v/Projects/pacup/pacup line 216.
Use of uninitialized value in string ne at /home/v/Projects/pacup/pacup line 216.
Use of uninitialized value in string ne at /home/v/Projects/pacup/pacup line 216.
    [>] Current: 0.10.0 ~> Latest: 0.10.0
[+] INFO: neovim is up to date
```

**Comment by oklopfer** [comment]
well u fixed the issue already

**Comment by ghost** [comment]
Not quite. Uno momento

**Comment by oklopfer** [comment]
> Not quite. Uno momento

oh true gunna have to turn it into an if check at this point

**Comment by oklopfer** [comment]
so much nesting

**Review by ghost — APPROVED** [review]

**Pull request merged**